src/components/views/CertificateForm.tsx

```diff
@@ -108,15 +108,26 @@ const CertificateForm: React.FC = () => {
   const handleResetFields = () => {
     setCertificate(INITIAL_CERTIFICATE);
   };
+  const handleSupplierOnSelect = (supplier: Supplier) => {
+    handleSupplierChange(supplier);
+    setShowSupplierLookup(false);
+  };
 
   return (
     <div className="new-cert-form">
+      {showSupplierLookup && (
+        <SupplierLookup
+          onClose={() => setShowSupplierLookup(false)}
+          onSupplierSelect={handleSupplierOnSelect}
```

Comment:
You should avoid using lamba functions directly here in callback. You should define these callback as constants wrapping them with 'useCallback' hook to cache them. Or even declare them out of the component. You do not want that browser declare them and initialize each time component re-renders, it's time consuming. Check this article:

https://jonhilton.net/typescript-and-react-forbidden-lambdas/